driver/tke_driver.go

```diff
@@ -793,3 +773,99 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
 	logrus.Info("unimplemented")
 	return nil
 }
+
+// RemoveLegacyServiceAccount remove any old service accounts that the driver has created
+func (d *Driver) RemoveLegacyServiceAccount(ctx context.Context, info *types.ClusterInfo) error {
+	//func (d *Driver) RemoveLegacyServiceAccount(ctx context.Context, info *types.ClusterInfo) error {
+	clientSet, err := d.getClientSet(ctx, info)
+	if err != nil {
+		return err
+	}
+
+	return util.DeleteLegacyServiceAccountAndRoleBinding(clientSet)
+}
+
+// getClientSet returns cluster clientSet
+func (d *Driver) getClientSet(ctx context.Context, info *types.ClusterInfo) (kubernetes.Interface, error) {
+
+	state, err := getState(info)
+	if err != nil {
+		return nil, err
+	}
+	svc, err := d.getTKEServiceClient(ctx, state, "GET")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := d.waitTKECluster(ctx, svc, state); err != nil {
```

Comment:
The waitTKECluster containers a infinity `for` loop. I think it should have timeout or something.